### ISSUE TITLE
"Restore & Delete" Issues

### ISSUE THREAD

**Issue by SinaBina** [YAFNET/YAFNET#742]
The [issue ](https://github.com/YAFNET/YAFNET/issues/740 )is still present.This time when you hit the "Delete All" Button, it only delete one record and not all record.on the other hand hitting "Delete All with 0 Post" throw the following error (Attached)

[Object reference not set to an instance of an object.zip](https://github.com/YAFNET/YAFNET/files/8748912/Object.reference.not.set.to.an.instance.of.an.object.zip)

To reproduce the error:
1-Hit "Delete All" and then hit "Delete All with 0 post"

**Comment by w8tcha** [comment]
Its impossible to follow your description. Please always describe the steps to reproduce the problem.

Did you click on delete All Posts or on topics? I tried both and it deletes all topics/post shown on the page.

**Comment by SinaBina** [comment]
This happens in an upgrade scenario from 1.9.6 using latest source.The following video may help:
[Bug.zip](https://github.com/YAFNET/YAFNET/files/8749222/Bug.zip)

**Comment by w8tcha** [comment]
Can you reproduce the issue with the latest source ? https://github.com/YAFNET/YAFNET/commit/bdbd6188e2abeedca78c5eb300a14c00f324d2d9

**Comment by SinaBina** [comment]
> Can you reproduce the issue with the latest source ? [bdbd618](https://github.com/YAFNET/YAFNET/commit/bdbd6188e2abeedca78c5eb300a14c00f324d2d9)

Yes.It is still there.

**Comment by w8tcha** [comment]
And is the stack trace still the same?

**Comment by SinaBina** [comment]
> And is the stack trace still the same?

Well...to me looks the same!
[Object reference not set to an instance of an object.zip](https://github.com/YAFNET/YAFNET/files/8756809/Object.reference.not.set.to.an.instance.of.an.object.zip)

**Comment by SinaBina** [comment]
> I committed a possible fix, is it throwing an error now?

The error has changed as you have added new try/catch.The first time I hit "Delete All" button, the following error is thrown:
[Error while deleting Message 12293.zip](https://github.com/YAFNET/YAFNET/files/8761547/Error.while.deleting.Message.12293.zip)
Then I refresh the page and hit the "Delete All" Button again, I get the following:
[Error while deleting Message 15754.zip](https://github.com/YAFNET/YAFNET/files/8761564/Error.while.deleting.Message.15754.zip)
And then again:
[Error while deleting Message 15771.zip](https://github.com/YAFNET/YAFNET/files/8761571/Error.while.deleting.Message.15771.zip)

I continued to repeat the task until there was no more error and all topics were deleted.This happens for both "Topics" and "Messages".

**Comment by w8tcha** [comment]
Too bad you deleted the messages, i would need the data of the message row(s) of Message ID 12293, 15754 and 15771

**Comment by SinaBina** [comment]
> Too bad you deleted the messages, i would need the data of the message row(s) of Message ID 12293, 15754 and 15771

No problem. I can restore the database.What information you need?

**Comment by w8tcha** [comment]
> > Too bad you deleted the messages, i would need the data of the message row(s) of Message ID 12293, 15754 and 15771
> 
> No problem. I can restore the database.What information you need?

I would need the db row in the yaf\_Message table with the Message Ids mentioned above

**Comment by SinaBina** [comment]
When I hit "Delete All" Button, I get 4 exceptions (while deleting 12293,15754,15,771,18123) and after that all messages are deleted.To make sure there is problem with messages, I restored the database, removed those rows manually and rerun yaf. This time I got 2 exceptions for 12292 and 15752. I repeated the above method and again I got two exception for 12288 and 15751.
I noticed that each time the message id number is decreased or go 1 record backward! 
So to me it is not related to the messages.

[Message_ID.zip](https://github.com/YAFNET/YAFNET/files/8790698/Message_ID.zip)
